pkg/cmd/admin/network/project_options.go

```diff
@@ -109,83 +97,82 @@ func (p *ProjectOptions) GetProjects() ([]*api.Project, error) {
 	}
 
 	errList := []error{}
-	projectList := []*api.Project{}
+	infoList := []*resource.Info{}
 	_ = r.Visit(func(info *resource.Info, err error) error {
 		if err != nil {
 			return err
 		}
-		project, ok := info.Object.(*api.Project)
+		_, ok := info.Object.(*kapi.Namespace)
 		if !ok {
-			err := fmt.Errorf("cannot convert input to Project: %v", reflect.TypeOf(info.Object))
+			err := fmt.Errorf("cannot convert input to Namespace: %v", reflect.TypeOf(info.Object))
 			errList = append(errList, err)
-			// Don't bail out if one project fails
+			// Don't bail out if one namespace fails
 			return nil
 		}
-		projectList = append(projectList, project)
+		infoList = append(infoList, info)
 		return nil
 	})
 	if len(errList) != 0 {
-		return projectList, kerrors.NewAggregate(errList)
+		return infoList, kerrors.NewAggregate(errList)
 	}
 
-	if len(projectList) == 0 {
-		return projectList, fmt.Errorf("No projects found")
+	if len(infoList) == 0 {
+		return infoList, fmt.Errorf("No projects found")
 	} else {
 		givenProjectNames := sets.NewString(p.ProjectNames...)
 		foundProjectNames := sets.String{}
-		for _, project := range projectList {
-			foundProjectNames.Insert(project.ObjectMeta.Name)
+		for _, info := range infoList {
+			ns, _ := info.Object.(*kapi.Namespace)
+			foundProjectNames.Insert(ns.ObjectMeta.Name)
 		}
 		skippedProjectNames := givenProjectNames.Difference(foundProjectNames)
 		if skippedProjectNames.Len() > 0 {
-			return projectList, fmt.Errorf("Projects %v not found", strings.Join(skippedProjectNames.List(), ", "))
+			return infoList, fmt.Errorf("Projects %v not found", strings.Join(skippedProjectNames.List(), ", "))
 		}
 	}
-	return projectList, nil
+	return infoList, nil
 }
 
-func (p *ProjectOptions) GetNetNamespaces() (*sdnapi.NetNamespaceList, error) {
-	netNamespaces, err := p.Oclient.NetNamespaces().List(kapi.ListOptions{})
```

Comment:
Yes, new client with old server will not work. Do we need this?
We want to obsolete NetNamespace resource at some point. We are supporting oadm pod-network command to work with both old/new clients against new servers.